evm/vm/base.py

```diff
@@ -636,10 +636,10 @@ def validate_block(self, block):
             )
 
         if block.is_genesis:
-            parent_header = None
+            validate_length_lte(block.header.extra_data, 32, title="BlockHeader.extra_data")
```

Comment:
Not to be fixed now, but this inline header validation bugs me.  It'd be nice to just have dedicated `validate_genesis_header` and `validate_chain_header` functions or something similar.  Otherwise we cannot support alternate genesis header formats without fully overriding this function.